env-setup/env-setup.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+set -e
+
+sof_test_dir=$(cd "$(dirname "$0")"; cd ..; pwd)
+
+function get_os_distro()
+{
+    grep -w ID /etc/os-release | awk -F= '{print $2}'
+}
+
+function die()
+{
+    printf "\e[31m%s\n\e[0m" "$*"
+    exit 1
+}
+
+function common_env_setup()
+{
+    printf "Add user %s to audio group\n" "$SUDO_USER"
+    if grep -q audio /etc/group | grep -q "$SUDO_USER"; then
+        printf "User \e[32m%s\e[0m is already in audio group\n" "$SUDO_USER"
+    else
+        usermod --append --groups audio "$SUDO_USER"
+    fi
+
+    printf "Enable NOPASSWD mode for user %s\n" "$SUDO_USER"
+    if grep -q "$SUDO_USER" /etc/sudoers; then
+        printf "NOPASSWD mode is already enabled for %s\n" "$SUDO_USER"
+    else
+        echo "$SUDO_USER ALL=(ALL:ALL) NOPASSWD:ALL" >> /etc/sudoers
+    fi
+
+    printf "Copy dynamic debug file 'sof-dyndbg.conf' to /etc/modprobe.d\n"
+    # Always use the latest file
+    cp sof-dyndbg.conf /etc/modprobe.d
+}
+
+function env_setup_ubuntu()
+{
+    printf "TODO"
+}
+
+function env_setup_fedora()
+{
+    local PKGS="jq expect alsa-utils-alsabat python3-graphviz python3-numpy \
+python3-scipy"
```

Comment:
The list of packages should be in a separate file so anyone can run a simple `dnf install $(cat fedora-requirements)` and ignore the rest of this script. I would not recommend to run this script because of some choices it makes and the silent modifications made to `/etc/modprobe.d/` and `/etc/sudoers`.

On the other hand the list of Fedora dependencies is not really debatable and it's very easy to uninstall some package if undesired.